app/src/androidTest/java/com/google/jetpackcamera/ImageCaptureDeviceTest.kt

```diff
@@ -0,0 +1,160 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Intent
+import android.net.Uri
+import android.os.Environment
+import android.provider.MediaStore
+import androidx.test.core.app.ActivityScenario
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import java.io.File
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class ImageCaptureDeviceTest {
+
+    @get:Rule
+    val cameraPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(
+            android.Manifest.permission.CAMERA,
+            android.Manifest.permission.READ_EXTERNAL_STORAGE,
+            android.Manifest.permission.WRITE_EXTERNAL_STORAGE
+        )
+
+    private val instrumentation = InstrumentationRegistry.getInstrumentation()
+    private var activityScenario: ActivityScenario<MainActivity>? = null
+    private val uiDevice = UiDevice.getInstance(instrumentation)
+
+    @Test
+    fun image_capture_default() = runTest {
+        val timeStamp = System.currentTimeMillis()
+        var directory = File(DIR_PATH)
+        val files = directory.listFiles()
+        activityScenario = ActivityScenario.launch(MainActivity::class.java)
+        uiDevice.wait(
+            Until.findObject(By.res("CaptureButton")),
+            5000
+        )
+        uiDevice.findObject(By.res("CaptureButton")).click()
+        uiDevice.wait(
+            Until.findObject(By.res("ImageCaptureSuccessToast")),
+            5000
+        )
+        val pictureTaken = (files.size + 1) == directory.listFiles().size
+        assert(pictureTaken)
+        if (pictureTaken) {
+            deleteFilesInDirAfterTimestamp(timeStamp)
+        }
+    }
+
+    @Test
+    fun image_capture_external() = runTest {
+        val timeStamp = System.currentTimeMillis()
+        var directory = File(DIR_PATH)
+        val files = directory.listFiles()
+        val launchIntent = Intent()
+        launchIntent.setAction(MediaStore.ACTION_IMAGE_CAPTURE)
+        launchIntent.putExtra(MediaStore.EXTRA_OUTPUT, TEST_URI)
+        launchIntent.setComponent(
+            ComponentName("com.google.jetpackcamera", "com.google.jetpackcamera.MainActivity")
+        )
+        activityScenario = ActivityScenario.launchActivityForResult(launchIntent)
```

Comment:
The [documentation](https://developer.android.com/reference/androidx/test/core/app/ActivityScenario#launchActivityForResult(android.content.Intent)) for this method says we should be using `androidx.activity.result.ActivityResult` instead of this method and links to https://developer.android.com/training/basics/intents/result#test. Since JCA is supposed to be following best practices, we should be using the methods suggested there.

Reply:
I read through https://developer.android.com/training/basics/intents/result#test a couple times and I don't think it can be applied in our test. Our test is lacking a start activity to call launchActivityForResult(), which is what ActivityScenario is for. In the documentation, however, the new way to launch an activity for result is still predicated on having a starting ComponentActivity, with which you can call registerForActivityResult() to get a Launcher to launch an activity externally. In the Test section, the documentation simply shows a way to mock/override a result being returned by the launched activity. 

Since we are trying to do a device test here, we don't need to override the result. Instead, we need a way to launch JCA externally, and the examples in the documentation don't help us. I also couldn't find any examples on how to obtain a launcher in a test in other places

Reply:
done